controllers/glanceapi_controller.go

```diff
@@ -238,11 +238,6 @@ func (r *GlanceAPIReconciler) reconcileInit(
 	} else if (ctrlResult != ctrl.Result{}) {
 		return ctrlResult, nil
 	}
-
-	if pvc.GetPvc().Status.Phase != corev1.ClaimBound {
```

Comment:
LGTM.
This actually solves the issue I saw, and I guess it's safe enough removing this check given the low complexity of the logic that manages the `pvc` (where we don't have particular constraints).
This change also work in a `SoS` environment, and avoid us to change any storageClass parameter (mostly the `VolumeBindingMode` from `WaitForFirstConsumer` to `Immediate`). 
In particular, in a `SoS` environment where `CSO` manages and reconciles the storageClasses parameters,  it's not safe (and it's not even supported) moving the storageClass to an `unmanaged` status, hence this change is important to make sure we can use any other backend.

Here an example of the pvc, bound to the default storageclass,  and the glance pod running:

```
> oc get sc
NAME              PROVISIONER                    RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
csi-manila-ceph   manila.csi.openstack.org       Delete          Immediate              false                  2d15h
local-storage     kubernetes.io/no-provisioner   Delete          WaitForFirstConsumer   false                  24m
standard          kubernetes.io/cinder           Delete          WaitForFirstConsumer   true                   2d15h
standard-csi      cinder.csi.openstack.org       Delete          WaitForFirstConsumer   true                   2d15h
```

```
> oc get pvc
NAME                STATUS    VOLUME             CAPACITY   ACCESS MODES   STORAGECLASS    AGE
glance              Pending                                                standard-csi    13s
mariadb-openstack   Bound     local-storage003   10Gi       RWO,ROX,RWX    local-storage   109m
```

```
NAME                      READY   STATUS    RESTARTS   AGE
glance-647f54679b-xn89s   1/1     Running   0          5m34s
```